lib/Zonemaster/Backend/Validator.pm

```diff
@@ -7,36 +7,85 @@ use warnings;
 use 5.14.2;
 
 use Exporter qw( import );
+use File::Spec::Functions qw( file_name_is_absolute );
 use JSON::Validator::Joi;
 use Readonly;
+use Zonemaster::Engine::Net::IP;
 
 our @EXPORT_OK = qw(
+  untaint_abs_path
   untaint_engine_type
+  untaint_ip_address
+  untaint_host
+  untaint_ldh_domain
+  untaint_mariadb_database
+  untaint_mariadb_user
+  untaint_non_negative_int
+  untaint_password
+  untaint_postgresql_ident
+  untaint_strictly_positive_int
+  untaint_strictly_positive_millis
 );
 
 our %EXPORT_TAGS = (
     untaint => [
         qw(
+          untaint_abs_path
           untaint_engine_type
+          untaint_ip_address
+          untaint_host
+          untaint_ldh_domain
+          untaint_mariadb_database
+          untaint_mariadb_user
+          untaint_non_negative_int
+          untaint_password
+          untaint_postgresql_ident
+          untaint_strictly_positive_int
+          untaint_strictly_positive_millis
           )
     ],
 );
 
+# Does not check value ranges within the groups
 Readonly my $IPV4_RE => qr/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/;
+
+# Does not check the length and number of the hex groups, nor the value ranges in the IPv4 groups
 Readonly my $IPV6_RE => qr/^([0-9a-f]{1,4}:[0-9a-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)$|([0-9a-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})$/i;
```

Comment:
This regex does not accept `::1` which you're likely to put in the {MYSQL,POSTGRESQL}.host properties if you want to use IPv6 locally.

Reply:
Fixed.